web/styles/search.css

```diff
@@ -2,150 +2,232 @@
-        height: calc(var(--header-height) - 1px);
-        min-height: 30px;
+        margin-bottom: 0;
+    }
+
+    .search-input {
+        font-size: 14px;
+        /* override bootstrap padding for input[type="text"] */
         padding: 0;
-        padding-right: 40px;
         border: none;
+        outline: none;
         border-radius: 0;
         font-family: "Source Sans 3 VF", sans-serif;
-        font-weight: 300;
-        line-height: var(--header-height);
+        font-weight: 400;
+        height: 28px;
+        /* override bootstrap min-height for input[type="text"] */
+        min-height: 28px;
+        background: transparent;
         text-overflow: ellipsis;
         overflow: hidden;
         white-space: nowrap;
-        padding-left: 35px;
-        color: hsl(0deg 0% 33%);
-
-        &:focus {
-            outline: 0;
-        }
-
-        @media (width < $sm_min) {
-            vertical-align: text-bottom;
-        }
+        flex-grow: 1;
     }
 
-    #searchbox-input-container:focus {
-        box-shadow: inset 0 0 0 2px hsl(204deg 20% 74%);
+    #searchbox-input-container {
+        display: flex;
+        align-items: center;
+        /* override .input-append style from app_components.js */
+        letter-spacing: normal;
+        height: 28px;
     }
 
-    .search_close_button {
-        position: absolute;
-        right: 0;
-        top: 6px;
-        background: none;
-        border-radius: 0;
-        border: none;
-        height: 30px;
-        text-align: center;
-        padding: 4px;
-        color: inherit;
-        opacity: 0.5;
-        font-size: 18px;
-        box-shadow: none;
-        text-shadow: none;
-        z-index: 5;
+    @media (width < $md_min) {
+        .navbar-search:not(.expanded) {
+            background: transparent;
+        }
 
-        &:hover {
+        .navbar-search:not(.expanded) .search_icon:hover {
             opacity: 1;
         }
+    }
 
-        &:disabled {
-            visibility: hidden;
+    @media (width < $sm_min) {
+        #searchbox_form {
+            margin: 0;
         }
 
-        @media (width < $sm_min) {
-            top: 0;
+        #searchbox-input-container,
+        .search-input {
+            height: var(--header-height);
         }
-    }
 
-    .search_icon {
-        position: absolute;
-        height: 100%;
-        text-decoration: none;
-        padding: 0 10px;
-        font-size: 20px;
-        z-index: 5;
-        left: 0;
-    }
-
-    #searchbox-input-container {
-        font-size: 90%;
-        letter-spacing: normal;
-        padding-left: 0;
+        #searchbox_form.expanded {
+            position: fixed;
+            left: 0;
+            right: 9px;
+            z-index: 1;
+            width: 100%;
+            border-radius: 0;
+        }
     }
-}
 
-.typeahead-menu li a {
-    .search_list_item {
+    .typeahead-menu {
         display: flex;
-        align-items: center;
+        flex-direction: column;
```

Comment:
Is there a reason to flex this with a column direction, and not just go with the normal document flow (`display: block`)? I'm not noticing a difference if I switch off `display: flex` on this in the Inspector, but maybe it's subtle? Or there's some case where we actually need flex properties?

Reply:
Oh interesting -- so I thought I needed this for the `search: Wrap long search suggestions.` commit, but maybe the flex wrap on `search_list_item` was enough. I also don't see any difference.